src/mps/mpo.jl

```diff
@@ -189,17 +189,26 @@ function linkindex(M::MPO,j::Integer)
   return li
 end
 
-function position!(M::MPO,
-                   j::Integer)
+function position!(M::MPO, j::Integer; kwargs...)
+  default_method = (rightLim(M) - leftLim(M) > 2) ? "qr" : "svd"
```

Comment:
Hi Katie,

Thanks for looking at this.

This logic seems a bit subtle here. Say we have the following situation:

-A-A-X-X-X-X-B-B-B-B-

(where A denotes a left orthogonal MPS tensor, B denotes a right orthogonal tensor, and X denotes a general tensor). Then we have that `leftLim(M) == 2`, `rightLim(M) == 7`. Say that someone calls `position!(M,10)`. Should this do a "qr" or an "svd"? While the gauge center is moved from sites 3 to 6, no truncation can occur, but after that it can. I am actually not really sure what is supposed to happen here (if "svd" is chosen, then between sites 3 to 6 we might be doing some truncation, but it is not right in general to truncate in that region).

Maybe Miles can chime in on how this situation is meant to be handled? It seems like the "proper" way to do it is that if `(rightLim(M) - leftLim(M) > 2)` then a separate `orthogonalize!` function should be called that orthogonalizes the region between `leftLim(M)` and `rightLim(M)`. Then after that, the gauge center can be moved with or without truncation (and "svd" or "qr" can be chosen freely by the user, with "svd" as the default since I believe it is more reliable).

Cheers,
Matt

Reply:
I was thinking something like this: we check to see whether any truncation parameters have been passed (maxdim, cutoff, ...) and if so, we use SVD. Otherwise we use QR since it is faster. I don't believe QR should blow up the bond dimension if done properly and on one site tensor at a time, correct? (I could be mistaken about this.)